### PR TITLE
Hardening suggestion on keyfolder

### PR DESCRIPTION
**Pull request by thesp0nge**
## What does this PR change?

This PR changes permissions on /srv/susemanager/salt/salt_ssh/temp_bootstrap_keys folder. Since SUMA store here private key files that are created using 0600 permission schema, also folder must be accessible only by salt user.

## GUI diff

No difference.

- [ X ] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [ X ] **DONE**

## Test coverage
- No tests: simple permission enforcement.
- 
- [ X ] **DONE**

## Links

None

- [ X ] **DONE**

## Changelogs

Fri Feb 18 09:18:51 UTC 2022 - Paolo Perego <paolo.perego@suse.com>

- Hardening suggestion: permissions for
  /srv/susemanager/salt/salt_ssh/temp_bootstrap_keys directory can be 0700
  instead of 0755. In this directory SUMA will store ssh private key files and
  it will create them with 0700 permission, so the directory itself can have
  the same.

### CODE REVIEW

File: java/spacewalk-java.spec [hunk 1]
```diff
@@ -706,7 +706,7 @@ chown tomcat:%{apache_group} /var/log/rhn/gatherer.log
 %dir %{_localstatedir}/lib/spacewalk
 %defattr(644,tomcat,tomcat,775)
 %attr(775, %{salt_user_group}, %{salt_user_group}) %dir %{serverdir}/susemanager/salt/salt_ssh
-%attr(775, %{salt_user_group}, %{salt_user_group}) %dir %{serverdir}/susemanager/salt/salt_ssh/temp_bootstrap_keys
+%attr(700, %{salt_user_group}, %{salt_user_group}) %dir %{serverdir}/susemanager/salt/salt_ssh/temp_bootstrap_keys
```
Comment (mcalmer) [inline java/spacewalk-java.spec:709]: I fear that tomcat needs at least to "read" the files in this directory. We have code which check for files exist and call cleanup to remove them after bootstrapping is done. That remove is called via SALT API, so I think write access is not needed.

File: java/spacewalk-java.changes [hunk 2]
```diff
@@ -1,3 +1,12 @@
+-------------------------------------------------------------------
+Fri Feb 18 09:18:51 UTC 2022 - Paolo Perego <paolo.perego@suse.com>
+
```
Comment (mcalmer) [inline java/spacewalk-java.changes:3]:
```suggestion
```

Comment (mcalmer) [reply java/spacewalk-java.changes:3]: The header is added automatically when we submit

**Review by mcalmer — CHANGES_REQUESTED** [review]
Doing the same changes in multiple PRs does not help us much to keep the overview.
Please close all duplicate PRs like https://github.com/uyuni-project/uyuni/pull/4464 
but please also add the required changes here to this PR.

There is an apache module missing for the proxy when we want to use the new headers.
See https://github.com/uyuni-project/uyuni/pull/4464#issuecomment-983642508

**Comment by thesp0nge** [comment]
Hi @mcalmer, sorry my fault. I wanted to submit a different PR without ASTS changes

**Pull request closed without merging**